### PR TITLE
Fixed Broken links present in Cross Site Scripting (XSS) page

### LINKED ISSUE #71: 404 for all XSS testing resources
See the latest OWASP Testing Guide article on how to test for the various kinds of XSS vulnerabilities.

Testing_for_Reflected_Cross_site_scripting_(OWASP-DV-001)
Testing_for_Stored_Cross_site_scripting_(OWASP-DV-002)
Testing_for_DOM-based_Cross_site_scripting_(OWASP-DV-003)

all of those links go to 404s.

### PR DESCRIPTION
**Pull request by BrunoLad**
- Fixed Types of XSS, Content Spoofing, Dom Based XSS Attacks broken
- Fixed all broken testing for XSS related links
- Fixed both quote Projects (ESAPI and Webgoat) broken links
- Fixed Filter Evasion Cheatsheet broken link

Fixes #71

### CODE REVIEW

File: pages/attacks/Cross-site-Scripting-(XSS).md [hunk 1]
```diff
@@ -28,7 +28,7 @@ cookies, session tokens, or other sensitive information retained by the
 browser and used with that site. These scripts can even rewrite the
 content of the HTML page. For more details on the different types of XSS
 flaws, see: [Types of Cross-Site
-Scripting](Types_of_Cross-Site_Scripting "wikilink").
+Scripting](../../Types_of_Cross-Site_Scripting "wikilink").
```
Comment (kingthorin) [inline pages/attacks/Cross-site-Scripting-(XSS).md:31]:
```suggestion
Scripting](../../Types_of_Cross-Site_Scripting).
```

File: pages/attacks/Cross-site-Scripting-(XSS).md [hunk 2]
```diff
@@ -88,8 +88,8 @@ under the guise of the vulnerable site.
 
 XSS attacks can generally be categorized into two categories: stored and
 reflected. There is a third, much less well-known type of XSS attack
-called [DOM Based XSS](DOM_Based_XSS "wikilink") that is discussed
-separately [here](DOM_Based_XSS "wikilink").
+called [DOM Based XSS](../DOM_Based_XSS "wikilink") that is discussed
```
Comment (kingthorin) [inline pages/attacks/Cross-site-Scripting-(XSS).md:91]:
```suggestion
called [DOM Based XSS](../DOM_Based_XSS) that is discussed
```

File: pages/attacks/Cross-site-Scripting-(XSS).md [hunk 3]
```diff
@@ -88,8 +88,8 @@ under the guise of the vulnerable site.
...
+separately [here](../DOM_Based_XSS "wikilink").
```
Comment (kingthorin) [inline pages/attacks/Cross-site-Scripting-(XSS).md:92]:
```suggestion
separately [here](../DOM_Based_XSS).
```

File: pages/attacks/Cross-site-Scripting-(XSS).md [hunk 4]
```diff
@@ -116,19 +116,19 @@ referred to as Non-Persistent or Type-II XSS.
 ### Other Types of XSS Vulnerabilities
 
 In addition to Stored and Reflected XSS, another type of XSS, [DOM Based
-XSS](DOM_Based_XSS "wikilink") was identified by [Amit Klein
+XSS](../DOM_Based_XSS "wikilink") was identified by [Amit Klein
```
Comment (kingthorin) [inline pages/attacks/Cross-site-Scripting-(XSS).md:119]:
```suggestion
XSS](../DOM_Based_XSS) was identified by [Amit Klein
```

File: pages/attacks/Cross-site-Scripting-(XSS).md [hunk 5]
```diff
@@ -116,19 +116,19 @@ referred to as Non-Persistent or Type-II XSS.
...
 in 2005](http://www.webappsec.org/projects/articles/071105.shtml). OWASP
 recommends the XSS categorization as described in the OWASP Article:
 [Types of Cross-Site
-Scripting](Types_of_Cross-Site_Scripting "wikilink"), which covers all
+Scripting](../../Types_of_Cross-Site_Scripting "wikilink"), which covers all
```
Comment (kingthorin) [inline pages/attacks/Cross-site-Scripting-(XSS).md:123]:
```suggestion
Scripting](../../Types_of_Cross-Site_Scripting), which covers all
```

File: pages/attacks/Cross-site-Scripting-(XSS).md [hunk 6]
```diff
@@ -116,19 +116,19 @@ referred to as Non-Persistent or Type-II XSS.
...
 these XSS terms, organizing them into a matrix of Stored vs. Reflected
 XSS and Server vs. Client XSS, where DOM Based XSS is a subset of Client
 XSS.
 
 #### XSS Attack Consequences
 
 The consequence of an XSS attack is the same regardless of whether it is
-stored or reflected ([or DOM Based](DOM_Based_XSS "wikilink")). The
+stored or reflected ([or DOM Based](../DOM_Based_XSS "wikilink")). The
```
Comment (kingthorin) [inline pages/attacks/Cross-site-Scripting-(XSS).md:131]:
```suggestion
stored or reflected ([or DOM Based](../DOM_Based_XSS)). The
```

File: pages/attacks/Cross-site-Scripting-(XSS).md [hunk 7]
```diff
@@ -143,7 +143,7 @@ modify a press release or news item could affect a company’s stock price
 or lessen consumer confidence. An XSS vulnerability on a pharmaceutical
 site could allow an attacker to modify dosage information resulting in
 an overdose. For more information on these types of attacks see
-[Content_Spoofing](Content_Spoofing "wikilink").
+[Content_Spoofing](../Content_Spoofing "wikilink").
```
Comment (kingthorin) [inline pages/attacks/Cross-site-Scripting-(XSS).md:146]:
```suggestion
[Content_Spoofing](../Content_Spoofing).
```

File: pages/attacks/Cross-site-Scripting-(XSS).md [hunk 8]
```diff
@@ -225,7 +225,7 @@ found in RFC 2397
     CONTENT="0;url=data:text/html;base64,PHNjcmlwdD5hbGVydCgndGVzdDMnKTwvc2NyaXB0Pg">
 
 These and others examples can be found at the OWASP [XSS Filter Evasion
-Cheat Sheet](XSS_Filter_Evasion_Cheat_Sheet "wikilink") which is a true
+Cheat Sheet](../../xss-filter-evasion-cheatsheet "wikilink") which is a true
```
Comment (kingthorin) [inline pages/attacks/Cross-site-Scripting-(XSS).md:228]:
```suggestion
Cheat Sheet](../../xss-filter-evasion-cheatsheet) which is a true
```

Comment (BrunoLad) [reply pages/attacks/Cross-site-Scripting-(XSS).md]: Solved threads in the latest commit.

**Pull request merged**